### DISCUSSION TITLE
How to run prisma migrate from code or without `.env`?

### DISCUSSION THREAD

**Discussion by ivan-shaban** [prisma/prisma#19672] [Q&A]
Hello, i want to work with different environments and has next flow:
1. build docker container with my app once
2. generate json file with env variables for concrete environment and run my container, this step can be repeated for every environment

So my question is - how properly run migrations and other db managements operations via cli at CI layer, when my env variables not stored into `.env` files (i use https://github.com/node-config/node-config lib for example), how to pass \ convert that prisma's cli could understand them?

I have access to all env variables values in my code at runtime, but i don't get how run them with cli, i understand that my concrete solution may be unique, but whole flow i as understand a kind of regular.

Thank you!

**Comment by ludralph** [comment]
Hi @ivan-shaban 👋 

Thank you for raising this question. It is currently not possible to run Prisma Migrate commands by code but we have two open feature requests for this:

- https://github.com/prisma/prisma/issues/13549 
- https://github.com/prisma/prisma/issues/4703

It would be great if you could leave a 👍 and a comment with your exact use case to help us prioritize this! 

There are some workarounds suggested in the open issues that you can take a look at to see if it helps with your use case.

- https://github.com/prisma/prisma/issues/13549#issuecomment-1144450394
- https://github.com/prisma/prisma/issues/4703#issuecomment-795204422
- https://github.com/prisma/prisma/issues/13549#issuecomment-1432959733

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by janpio** [comment]
> how properly run migrations and other db managements operations via cli at CI layer, when my env variables not stored into `.env` files (i use [node-config/node-config](https://github.com/node-config/node-config) lib for example), how to pass \ convert that prisma's cli could understand them?
> 
> I have access to all env variables values in my code at runtime, but i don't get how run them with cli, i understand that my concrete solution may be unique, but whole flow i as understand a kind of regular.

Environment variables are an actual thing - variables in the environment of a program. The solution you describe is actually "just" configuration via some file that is then loaded into a program. Does it actually also load them into `process.env` in Node or just via `config`?

Assuming you want to use Prisma CLI in your CI, you will need to find a way to transport the values from your solution to the real environment (or a temporary `.env` file) so the CLI can pick them up.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
